src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,293 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
-
-    switch (aPath.mAttributeId)
+    switch (request.path.mAttributeId)
     {
+    case IcdManagement::Attributes::ClusterRevision::Id:
+        return aEncoder.Encode(kRevision);
+
+    case IcdManagement::Attributes::FeatureMap::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetFeatureMap());
+
     case IcdManagement::Attributes::IdleModeDuration::Id:
-        return ReadIdleModeDuration(aPath.mEndpointId, aEncoder);
+        return aEncoder.Encode(mICDConfigurationData.GetIdleModeDuration().count());
 
     case IcdManagement::Attributes::ActiveModeDuration::Id:
-        return ReadActiveModeDuration(aPath.mEndpointId, aEncoder);
+        return aEncoder.Encode(mICDConfigurationData.GetActiveModeDuration().count());
 
     case IcdManagement::Attributes::ActiveModeThreshold::Id:
-        return ReadActiveModeThreshold(aPath.mEndpointId, aEncoder);
+        return aEncoder.Encode(mICDConfigurationData.GetActiveModeThreshold().count());
 
-    case IcdManagement::Attributes::FeatureMap::Id:
-        return ReadFeatureMap(aPath.mEndpointId, aEncoder);
 #if CHIP_CONFIG_ENABLE_ICD_LIT
     case IcdManagement::Attributes::OperatingMode::Id:
-        return ReadOperatingMode(aPath.mEndpointId, aEncoder);
+        return ReadOperatingMode(aEncoder);
 #endif // CHIP_CONFIG_ENABLE_ICD_LIT
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-    case IcdManagement::Attributes::RegisteredClients::Id:
-        return ReadRegisteredClients(aPath.mEndpointId, aEncoder);
 
-    case IcdManagement::Attributes::ICDCounter::Id:
-        return ReadICDCounter(aPath.mEndpointId, aEncoder);
+    case IcdManagement::Attributes::UserActiveModeTriggerHint::Id:
+        return aEncoder.Encode(mUserActiveModeTriggerBitmap);
 
-    case IcdManagement::Attributes::ClientsSupportedPerFabric::Id:
-        return ReadClientsSupportedPerFabric(aPath.mEndpointId, aEncoder);
+    case IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id:
+        return aEncoder.Encode(
+            CharSpan(mUserActiveModeTriggerInstruction,
+                     strnlen(mUserActiveModeTriggerInstruction, IcdManagement::kUserActiveModeTriggerInstructionMaxLength)));
 
-    case IcdManagement::Attributes::MaximumCheckInBackOff::Id:
-        return ReadMaximumCheckInBackOff(aPath.mEndpointId, aEncoder);
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
     }
-
-    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ICDManagementAttributeAccess::ReadIdleModeDuration(EndpointId endpoint, AttributeValueEncoder & encoder)
+CHIP_ERROR ICDManagementCluster::Attributes(const ConcreteClusterPath & path,
+                                            ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
 {
-    return encoder.Encode(mICDConfigurationData->GetIdleModeDuration().count());
+    AttributeListBuilder attributeListBuilder(builder);
+    const DataModel::AttributeEntry kMandatoryAttributes[] = {
```

Comment:
Please use `kMandatoryMetadata` from this cluster's Metadata.h instead of duplicating it.